src/praisonai-mobile/engines/src/assert-ledger.test.ts

```
@@ -0,0 +1,99 @@
+  counted.ok(true);
+  assert.equal(l.made(), 2);
+});
+
+test("a BARE assert(x) is counted too", () => {
+  // `assert` is both callable and a namespace of callables. Counting only the
+  // namespace leaves a whole style of assertion invisible to the ledger --
+  // measured: dropping the apply trap's counter survived the entire suite.
+  const l = ledger();
+  const counted: Ledger["assert"] = l.assert;
+  counted(true);
+  counted(1 === 1);
```

Biome (2.5.7)

[error] This comparison uses the same expression on both sides.

(lint/suspicious/noSelfCompare)